### ISSUE TITLE
Simplifying zero-like expression discards Integer/Float type

### ISSUE THREAD

**Issue by clinssen** [sympy/sympy#21740]
Simplifying a zero-like expression loses information as to whether the original expression was integer-like or float-like.

This behaviour changed somewhere between sympy 1.4 and 1.8; more precisely, it is caused by commit 7649a7c3f1c6c4514b6b1ceaaffede5634673867.

Compare:

**Before:**
```Python
>>> sympy.simplify(sympy.Float("0."))
0.0
>>> type(sympy.simplify(sympy.Float("0.")))
sympy.core.numbers.Float
```

**After:**
```Python
>>> sympy.simplify(sympy.Float("0."))
0
>>> type(sympy.simplify(sympy.Float("0.")))
sympy.core.numbers.Zero
```

Note that this issue does not occur for One, because there is no special case in simplify() to return One for a one-like expression (as was introduced in 7649a7c3f1c6c4514b6b1ceaaffede5634673867 for Zero):

```Python
>>> sympy.simplify(sympy.Float("1."))
1.00000000000000
>>> type(sympy.simplify(sympy.Float("1.")))
sympy.core.numbers.Float
```

This is in any case inconsistent; a solution should address both the Zero and the One case in a consistent manner.

Potential solutions:
- Should zero-like expressions not be simplified to Zero, but remain a Float or Integer? (Basically undo the special case introduced by 7649a7c3f1c6c4514b6b1ceaaffede5634673867.)
- Should a new flag be added to the Zero class (which currently inherits from IntegerConstant) to indicate integer-ness or float-ness?

Potentially related discussion: #6545

**Comment by smichr** [comment]
I'll see if this is resolved in #21619 and add this in if it is not.

**Comment by clinssen** [comment]
@smichr: many thanks!
